lumicks/pylake/kymotracker/detail/trace_line_2d.py

```diff
@@ -180,14 +182,23 @@ def _traverse_line_direction(
     return nodes
 
 
+@dataclass(frozen=True)
 class KymoLineData:
-    __slots__ = ["time_idx", "coordinate_idx"]
+    """Data class which holds the raw data for a Kymograph line.
+
+    Parameters
+    ----------
+    time_idx : List[float]
+        List of time indices on the tracked image [pixels]
+    coordinate_idx : List[float]
+        List of coordinate indices on the tracked image [pixels]
+
+    Note: Some of the kymotracker functions return numpy arrays. Appending too many small numpy
+    arrays was a bottleneck and therefore we explicitly use lists."""
```

Comment:
The usual Python convention is to end a docstring with `"""` by itself on the last line:
```suggestion
    Note: Some of the kymotracker functions return numpy arrays. Appending too many small numpy
    arrays was a bottleneck and therefore we explicitly use lists.
    """
```